server/data/educationAndWorkPlanClient.test.ts

```diff
@@ -40,14 +41,14 @@ describe('educationAndWorkPlanClient', () => {
       const prisonNumber = 'A1234BC'
       const systemToken = 'a-system-token'
 
-      const createGoalsRequest = aValidCreateGoalsRequestWitMultipleGoals(prisonNumber)
+      const createGoalsRequest = aValidCreateGoalsRequestWitMultipleGoals()
       const expectedResponseBody = {}
       educationAndWorkPlanApi
-        .post(`/action-plans/${prisonNumber}/goals`, createGoalsRequest)
+        .post(`/action-plans/${prisonNumber}/goals`, requestBody => isEqual(requestBody, createGoalsRequest))
```

Comment:
I was in this test file anyway because I needed to correct the calls to the test data builders and the call to the `createGoals` client method (based on the main changes in this PR)

But intellij was highlighting errors on all of the calls to `nock` that use a request body. EG: this original line:
```
 educationAndWorkPlanApi
        .post(`/action-plans/${prisonNumber}/goals`, createGoalsRequest)
```
intellij highlighted an error on that line because the request body argument (`createGoalsRequest`) does not match  one of the supported method signatures:
![Screenshot 2024-07-23 at 09 25 10](https://github.com/user-attachments/assets/aa84c0b3-739f-4a35-a5ed-b711c04a1cbe)

Clicking into the `post` method it defines the method sig as:
```
  type InterceptFunction = (
    uri: string | RegExp | { (uri: string): boolean },
    requestBody?: RequestBodyMatcher,
    interceptorOptions?: Options,
  ) => Interceptor
```
and clicking into the `RequestBodyMatcher` type it is defined as:
```
  type RequestBodyMatcher =
    | string
    | Buffer
    | RegExp
    | DataMatcherArray
    | DataMatcherMap
    | { (body: any): boolean }
```

We need to provide an arg that fulfils the last one - a function that takes the received request body as an argument and returns a boolean.
(This is why I've brought in `lodash` in order to get the `isEqual` function - happy to use something else if there is already something suitable in our dependencies)

FWIW I believe these calls to `nock` worked previously by virtue of javascript objects being truthy! 🤮